### PR TITLE
guess at setting doc meta title alternative way

### PR DESCRIPTION
**Pull request by darrentayloruk**
http://docutils.sourceforge.net/docs/ref/rst/directives.html#metadata-document-title

### CODE REVIEW

File: source/cloud/vault/connecting.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ```eval_rst
+.. title: UKFast Documentation | eCloud Vault | Connecting to eCloud Vault
```
Comment (laf) [inline source/cloud/vault/connecting.md:2]: If this is to set the actual title of the page then you can just edit this line:

`# Connecting to eCloud Vault`

**Review by laf — DISMISSED** [review]

**Comment by garryprior** [comment]
does this mean that the page title will automatically be set as the meta title?  i.e. no need for the person writing / editing the page to add in the meta title manually?

will it also check it's unique on docs.ukfast?

**Comment by garryprior** [comment]
also what about index (.rst) files where the page title looks like

```
=========
eCloud Vault
=========
```
rather than 

```
# eCloud Vault
```

can we still automate this being added as the meta title?

**Comment by darrentayloruk** [comment]
I think this needs to be separate, to avoid dupes and long titles, as the page title can be something clear to the use

e.g.

https://docs.ukfast.co.uk/desktop/fastdesk/icantseemylocaldrivesanddevicesinfastdesk.html

comes up as a too long meta title.. but the title is OK, just needs a shorter meta title

Similarly ones with just one word titles

https://docs.ukfast.co.uk/desktop/fastdrive/index.html

Are perfectly fine to have as h1, but have a high chance of being a duplicate with something else

File: source/cloud/vault/connecting.md (hunk 1 above)
Comment (darrentayloruk) [reply source/cloud/vault/connecting.md:2]: its the meta title that goes in  <title>xxx</title> looking to change that seems linked to the title set by # as above example at the moment..

**Comment by laf** [comment]
> does this mean that the page title will automatically be set as the meta title? i.e. no need for the person writing / editing the page to add in the meta title manually?

@garryprior No, the title and meta title are two different things. Meta information needs to be set as per the recent PR to work but the normal page title can just be updated within the markdown.

**Comment by garryprior** [comment]
OK thanks.  
So I am currently editing https://docs.ukfast.co.uk/cloud/flex/ipv6.html 

https://github.com/garryprior/docs.ukfast.co.uk-GARRY/blob/9c76eb33a9d7a7a0e238868a64ee14f0a4fc2362/source/cloud/flex/ipv6.md

I have added this to the top of the page, this is correct yes?

```
```eval_rst
.. meta::
   :title: UKFast Documentation | eCloud Flex | Using IPv6 on eCloud Flex
   :description: How to enable and use IPv6 on eCloud Flex - Linux and Windows
```

**Comment by darrentayloruk** [comment]
@laf @garryprior 

Yeah so just to clarify.. (and dunno if you saw my email if makes more sense), the meta markdown update is adding a meta name="title".. however its not updating the <title></title> tags which is what we are after :)

I had a read and think this change should change the value in <title> but leave the h1 as it is..

**Comment by ghost** [comment]
Tried modifying the template with:

```
    <title>
        {% if meta is defined %}
        {% if meta.title is defined %}
        {{ meta.title }}
        {% else %}
        Nope
        {% endif %}
        {% else %}
        YAY
        {% endif %}
    </title>
```

But it produces the following:

`<title>Nope</title>`

Which suggests the meta value exists, but meta.title is undefined.

Am I doing anything wrong here? I'm fairly new to jinja to be honest but it looks correct.

Also please ignore the formatting, just easier to read while testing! :)

**Comment by laf** [comment]
You might be on to a loser with this @BenWUKFast: http://docutils.sourceforge.net/docs/ref/rst/directives.html#meta indicates that meta tags are strings only and not an array to loop through.

**Comment by darrentayloruk** [comment]
thanks for look @BenWUKFast !

see in my first post here I found something about setting the title.. so can we try to see if you can set and catch it using that?

.. title: UKFast Documentation | eCloud Vault | Connecting to eCloud Vault

can you see if that works, instead of doing meta.title in the code it would just be title?

**Comment by garryprior** [comment]
how is this progressing?

@BenWUKFast have you had chance to investigate further?

**Comment by ghost** [comment]
@garryprior Spoke with @laf about this and it doesn't look like there's an easy way to expose the meta variables to the template. We think it will need an extension written. Harry G is learning python at the moment and he has reluctantly volunteered through no coercion from me or @laf to give it a go. He's gonna catch up with me at some point to see exactly what needs doing but has the gist.

**Comment by darrentayloruk** [comment]
did anyone try the title thing from my last comment?

**Comment by garryprior** [comment]
has this progressed at all @BenWUKFast ?

**Comment by ghost** [comment]
I'll ask if Harry has had chance to write an extension. He's not mentioned it so I doubt it.

**Comment by garryprior** [comment]
@darrentayloruk @Xiol think this should work so lets give it a go.  Will roll back if not.

```
Can you approve and push merge request

https://github.com/ukfast/docs.ukfast.co.uk/pull/197/

?

I;ve asked Dane and he thinks a pythonm submodule shouldn't be needed and this mayb work... we can just try and see.. and if problem just remove it?
```

**Review by garryprior — APPROVED** [review]

**Pull request merged**